src/OpenTelemetry.Api/Internal/BlankSpan.cs

```diff
@@ -66,21 +65,11 @@ public void SetAttribute(string key, double value)
         {
         }
 
-        /// <inheritdoc />
-        public void SetAttribute(string key, bool value)
-        {
-        }
-
         /// <inheritdoc />
         public void AddEvent(string name)
         {
         }
 
-        /// <inheritdoc />
-        public void AddEvent(string name, IDictionary<string, object> attributes)
```

Comment:
by removing it - an option to `AddEvent` optimization that prevents allocation of an `Event` object disappears. So API forces allocation. Is it intended?

Reply:
Also it makes the constructor that accepts DateTimeOffset more visible. Which can lead to the incorrect use of an API where the timestamp is passed explicitly.

Reply:
there is a requirement to support lazy events. This also does not prevent allocation (lazy is Func which is usually an allocation). 

I'm still strong that instrumentation that cares about perf MUST use IsRecording flag to optimize. Creating a dictionary of attributes is bad enough before checking for isRecording so it does not make this method optimal anyway.

Preventing allocations by exposing methods on Span mans that Span API explodes to support all possible/popular ways to create events/links/set attributes. I would prefer to stay on the small and clean API (which requires instrumentation to optimize by using  IsRecording)

Reply:
> Also it makes the constructor that accepts DateTimeOffset more visible. Which can lead to the incorrect use of an API where the timestamp is passed explicitly.

Let's do something about it and make event timestamp usage hard